sky/engine/core/painting/RRect.dart

```diff
@@ -1,32 +1,234 @@
+
+  /// Whether this rounded rectangle has no side with a straight section.
+  bool get isEllipse => width <= 2 * radiusX && height <= 2 * radiusY;
+
+  /// Whether this rounded rectangle would draw as a circle.
+  bool get isCircle => width == height && isEllipse;
+
+  /// The lesser of the magnitudes of the width and the height of this rounded
+  /// rectangle.
+  double get shortestSide {
+    double w = width.abs();
+    double h = height.abs();
+    return w < h ? w : h;
+  }
+
+  /// The point halfway between the left and right and the top and bottom edges of this rectangle.
+  Point get center => new Point(left + width / 2.0, top + height / 2.0);
+
+  /// Whether the given point lies inside the rounded rectangle.
+  bool contains(Point point) {
+    if (point.x < left || point.x >= right || point.y < top || point.y >= bottom)
+      return false; // outside bounding box
+    double leftInner = left + radiusX;
+    double rightInner = right - radiusX;
+    double topInner = top + radiusY;
+    double bottomInner = bottom - radiusY;
+    if (point.x >= leftInner && point.x <= rightInner)
+      return true; // inside tallMiddleRect
+    if (point.y >= topInner && point.y <= bottomInner)
+      return true; // inside wideMiddleRect
+    // it is in one of the corners
+    // convert this to a test of the unit circle
+    double x, y;
+    if (point.x > leftInner) {
+      assert(point.x > rightInner);
+      x = point.x - (rightInner - leftInner);
+    } else {
+      x = point.x;
+    }
+    if (point.y > topInner) {
+      assert(point.y > bottomInner);
+      y = point.y - (bottomInner - topInner);
+    } else {
+      y = point.y;
+    }
+    x = x / (radiusX * 2) - 0.5;
+    y = y / (radiusY * 2) - 0.5;
+    // check if the point is outside the unit circle
+    if (x * x + y * y > 0.25)
+      return false;
+    return true;
+  }
+
+  /// Linearly interpolate between two rounded rectangles.
+  ///
+  /// If either is null, this function substitutes [RRect.zero] instead.
+  static RRect lerp(RRect a, RRect b, double t) {
+    if (a == null && b == null)
+      return null;
+    if (a == null)
+      return new RRect.fromLTRBXY(b.left * t, b.top * t, b.right * t, b.bottom * t, b.radiusX * t, b.radiusY * t);
+    if (b == null) {
+      double k = 1.0 - t;
+      return new RRect.fromLTRBXY(a.left * k, a.top * k, a.right * k, a.bottom * k, a.radiusX * k, a.radiusY * k);
+    }
+    return new RRect.fromLTRBXY(
+      lerpDouble(a.left, b.left, t),
+      lerpDouble(a.top, b.top, t),
+      lerpDouble(a.right, b.right, t),
+      lerpDouble(a.bottom, b.bottom, t),
+      lerpDouble(a.radiusX, b.radiusX, t),
+      lerpDouble(a.radiusY, b.radiusY, t)
+    );
+  }
+
+  bool operator ==(dynamic other) {
+    if (identical(this, other))
+      return true;
+    if (other is! RRect)
+      return false;
+    final RRect typedOther = other;
+    for (int i = 0; i < 6; i += 1) {
```

Comment:
Done. I've also done it for Rect, which is where I copy-pasta'ed this code from. That seems like a relatively hot path, though. Do we want to go back to hard-coding the 4 in, at least? We presumably compare rects a lot.